src/course-outline/publish-modal/PublishModal.jsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable import/named */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  ModalDialog,
+  Button,
+  ActionRow,
+} from '@edx/paragon';
+import { useSelector } from 'react-redux';
+
+import { getCurrentSection } from '../data/selectors';
+import messages from './messages';
+
+const PublishModal = ({
+  isOpen,
+  onClose,
+  onPublishSubmit,
+}) => {
+  const intl = useIntl();
+  const { displayName, childInfo } = useSelector(getCurrentSection);
+  const subSections = childInfo?.children || [];
+
+  return (
+    <ModalDialog
+      className="publish-modal"
+      isOpen={isOpen}
+      onClose={onClose}
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header className="publish-modal__header">
+        <ModalDialog.Title>
+          {intl.formatMessage(messages.title, { title: displayName })}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+      <ModalDialog.Body>
+        <p className="small">{intl.formatMessage(messages.description)}</p>
+        {subSections.filter(subSection => subSection.hasChanges).map((subSection) => {
+          const units = subSection.childInfo.children.filter(unit => unit.hasChanges);
+
+          return units.length ? (
```

Comment:
Same question as section lists?

Reply:
It is same, 0 is considered a falsy value.